src/state_migration/tests/mod.rs

```diff
@@ -0,0 +1,103 @@
+// Copyright 2019-2023 ChainSafe Systems
+// SPDX-License-Identifier: Apache-2.0, MIT
+
+use crate::shim::state_tree::StateRoot;
+use crate::{
+    daemon::bundle::get_actors_bundle,
+    networks::{ChainConfig, Height, NetworkChain},
+    state_migration::run_state_migrations,
+};
+use anyhow::*;
+use cid::Cid;
+use fvm_ipld_encoding::CborStore;
+use pretty_assertions::assert_eq;
+use std::{str::FromStr, sync::Arc};
+
+#[tokio::test]
+async fn test_nv17_state_migration_calibnet() -> Result<()> {
+    // forest_filecoin::state_migration: State migration at height Shark(epoch 16800) was successful,
+    // Previous state: bafy2bzacedxtdhqjsrw2twioyaeomdk4z7umhgfv36vzrrotjb4woutphqgyg,
+    // new state: bafy2bzacecrejypa2rqdh3geg2u3qdqdrejrfqvh2ykqcrnyhleehpiynh4k4.
+    //
+    // See <https://github.com/ChainSafe/forest/actions/runs/5579505385/jobs/10195488001#step:6:232>
+    test_state_migration(
+        Height::Shark,
+        NetworkChain::Calibnet,
+        Cid::from_str("bafy2bzacedxtdhqjsrw2twioyaeomdk4z7umhgfv36vzrrotjb4woutphqgyg")?,
+        Cid::from_str("bafy2bzacecrejypa2rqdh3geg2u3qdqdrejrfqvh2ykqcrnyhleehpiynh4k4")?,
+    )
+    .await
+}
+
+// #[tokio::test]
```

Comment:
What do we do with these commented out tests?

Reply:
They are commented out because the state car files are maybe too large to commit (10MiB and 20MiB for nv18 and nv19 respectively)

Reply:
Could we put them in DO?

Reply:
Yes but the tests then rely on network which might not be the best either

Reply:
We already rely on the network when downloading actor bundles. Doing it once and storing it somewhere in a git-ignored directory is fine. What do you think about it? In general, we should avoid having dead code that is neither used nor even compiled.

Reply:
@lemmih Are you positive with uploading a few state car files to DO for these tests?

Reply:
@LesnyRumcajs Or maybe I can keep only the nv17 test as the input car is very small, leaving nv18 nv19 tests exercise for the `Test first development` section in the guide, does that work?

Reply:
@lemmih @LesnyRumcajs Thanks, after thinking twice, I deleted the nv18 and nv19 tests and leave them as exercises for the `Test first development` section. The functionality has been covered by `calibnet_migration_regression_tests.sh`, and the goal of this unit test is to facilitate development and troubleshooting, keeping nv17 test as a reference implementation should be sufficient.